### ISSUE TITLE
CAN SDO Map Value request can corrupt CAN map when save in progress

### ISSUE THREAD

**Issue by davefiddes** [jsphuebner/libopeninv#3]
Found this problem by code inspection when porting the libopeninv stm32_can.cpp to the C2000.

When an SDO Map Value Tx or RX request is being processed in `Can::ProcessSDO()` there is no interlock with the CAN map save operation. The isSaving interlock is used to stop the temporarily modified CAN maps from being used for sending and receiving. It seems that without the interlock weird things would happen with the processing of the new mapping request.

The correct behaviour would appear to be to return an SDO_ABORT with a new error code SDO_ERR_GENERAL (0x08000000).

**Comment by jsphuebner** [comment]
Yes, good point! Currently rewriting the module (see Zombieverter) but I think I'll also back port this to the current version

**Comment by jsphuebner** [comment]
Fixed with latest commit in float branch
